src/GTP.cpp

```diff
@@ -362,24 +372,59 @@ bool GTP::execute(GameState & game, std::string xinput) {
                 gtp_fail_printf(id, "syntax error");
                 return 1;
             }
+            if (analysis_output) {
+                cfg_analyze_interval_centis = interval;
+                if (id != -1) gtp_printf_raw("=%d\n", id);
+                else gtp_printf_raw("=\n");
+            }
             // start thinking
             {
                 game.set_to_move(who);
                 int move = search->think(who);
                 game.play_move(move);
 
                 std::string vertex = game.move_to_text(move);
-                gtp_printf(id, "%s", vertex.c_str());
+                if (!analysis_output) {
+                    gtp_printf(id, "%s", vertex.c_str());
+                } else {
+                    gtp_printf_raw("play %s\n", vertex.c_str());
+                }
             }
             if (cfg_allow_pondering) {
                 // now start pondering
                 if (!game.has_resigned()) {
                     search->ponder();
                 }
             }
+            if (analysis_output) {
+                gtp_printf_raw("\n");
```

Comment:
Can't this be merged with the previous call?

Also, it looks suspicious that your output depends on when the ponder search finishes?

Or is the idea that if the GUI issues lz-genmove_analyze, it should expect potential ponder output after the play?

Reply:
search->ponder() will print to the gtp stream as well (that is where all the info lines come from), adding to the multi-line response. That's also why you can't use the regular gtp_ calls, since they add \n\n which means end of response. But we don't want end of response until search->ponder returns. That is why there is a lone \n there.

Reply:
Might be good to add some minimal documentation about this in the code. Just pointing out that this will terminate a multi-line response by adding the 2nd newline to the newline in the last analysis output.